src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/NativeWrapper.cs

```diff
@@ -1272,14 +1272,14 @@ public static EventLogHandle ConvertToSafeHandle(UnsafeNativeMethods.EvtVariant
             IntPtr ptr = val.Reference;
             if (ptr == IntPtr.Zero)
             {
-                return Array.CreateInstance(typeof(T), 0);
+                return Array.Empty<T>();
             }
             else
             {
-                Array array = Array.CreateInstance(typeof(T), (int)val.Count);
+                T[] array = new T[val.Count];
                 for (int i = 0; i < val.Count; i++)
                 {
-                    array.SetValue(Marshal.PtrToStructure<T>(ptr), i);
+                    array[i] = Marshal.PtrToStructure<T>(ptr);
```

Comment:
```suggestion
                    array[i] = ((T*)ptr)[i];
```

Reply:
I was going to ask how this worked but then I saw the other suggestion around using the unmanaged generic constraint.

Reply:
Thanks. I've fixed this and the other array methods like it.